### PR TITLE
[Inspector] Prevent crashing when a mesh's name is of the wrong type.

### PR DESCRIPTION
**Pull request by carolhmj**
A user can mistakenly set a mesh's name with the wrong type (not string), like this example: https://playground.babylonjs.com/#XT7L5S, where trying to open the Inspector makes it crash. This PR just adds some extra type checking to avoid it crashing, and also adds a quick check to warn the user when adding to a scene.

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

File: packages/dev/inspector/src/components/actionTabs/tabs/propertyGrids/parentPropertyGridComponent.tsx [hunk 1]
```diff
@@ -22,14 +22,23 @@ export class ParentPropertyGridComponent extends React.Component<IParentProperty
...
 
         const sortedNodes = scene
             .getNodes()
             .filter((n) => n !== node)
-            .sort((a, b) => (a.name || "no name").localeCompare(b.name || "no name"));
+            .sort((a, b) => {
+                const aName = this._getNameForSorting(a);
+                const bName = this._getNameForSorting(b);
+                return aName.localeCompare(bName);
+            });
```
Comment (RaananW) [inline packages/dev/inspector/src/components/actionTabs/tabs/propertyGrids/parentPropertyGridComponent.tsx:41]:
```suggestion
            .map((n) => this._getNameForSorting).
            .sort((a, b) => {
                return aName.localeCompare(bName);
            });
```

optimized :-)

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/13893/merge

Test environment: 
https://babylonsnapshots.z22.web.core.windows.net/refs/pull/13893/merge/index.html

To test a playground add it to the URL, for example:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/13893/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/13893/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/13893/merge
https://gui.babylonjs.com/?snapshot=refs/pull/13893/merge
https://nme.babylonjs.com/?snapshot=refs/pull/13893/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/13893/merge#BCU1XR#0

File: packages/dev/inspector/src/components/actionTabs/tabs/propertyGrids/parentPropertyGridComponent.tsx [hunk 2]
```diff
@@ -57,7 +65,7 @@ export class ParentPropertyGridComponent extends React.Component<IParentProperty
                     noDirectUpdate={true}
                     onSelect={(value) => {
                         const nodeAsTransform = node as TransformNode;
-                        if (value < 0) {
+                        if (typeof value === "number" && value < 0) {
```
Comment (Popov72) [inline packages/dev/inspector/src/components/actionTabs/tabs/propertyGrids/parentPropertyGridComponent.tsx:68]: Don't you want to check `typeof value !== "number" || value < 0` instead? Else, for a non number value (only other possible value is "string" I guess?), you will take the `else` branch and do `sortedNodes[value as number]`, which won't work if value is not a number.

File: packages/dev/inspector/src/components/actionTabs/tabs/propertyGrids/parentPropertyGridComponent.tsx [hunk 3]
```diff
@@ -34,9 +34,8 @@ export class ParentPropertyGridComponent extends React.Component<IParentProperty
         const sortedNodes = scene
             .getNodes()
             .filter((n) => n !== node)
+            .map((n) => this._getNameForSorting).
```
Comment (RaananW) [inline packages/dev/inspector/src/components/actionTabs/tabs/propertyGrids/parentPropertyGridComponent.tsx:37]: one dot too much... I take SOME of the responsibility here ;-)

File: packages/dev/inspector/src/components/actionTabs/tabs/propertyGrids/parentPropertyGridComponent.tsx (hunk 2 above)
Comment (carolhmj) [reply packages/dev/inspector/src/components/actionTabs/tabs/propertyGrids/parentPropertyGridComponent.tsx:68]: Good catch, thanks!

**Review by sebavan — CHANGES_REQUESTED** [review]
I am not sure we should do this. This is adding extra code and why doing it here and not for all the other APIs we have ?

I guess the OP of the forum thread should use typescript to not use the API incorrectly ?

There might as well be smthg I did not get.

**Comment by Popov72** [comment]
> I am not sure we should do this. This is adding extra code and why doing it here and not for all the other APIs we have ?

This is not the first time this problem has been reported to us and it's not always easy to find that it comes from a bad type in the mesh names, so dealing with the problem on our side would be nice, but as you say, why do it only for this one and not for everything else that could go wrong... It's a bug on the user side, after all, not in Babylon.js. So, after a second though, I would also vote not to do it.

**Comment by RaananW** [comment]
I don't see a warning in the scene as something critical, but I do think the solution for the inspector is important.

This happens from time to time. We can stay defensive, at least in our tools. This is obviously not "our issue", but warning the user they did something wrong is not a bad thing :-)

If you feel it's not the right place, maybe remove the scene changes. The playground changes should stay, as this breaks the inspector.

**Comment by sebavan** [comment]
Love the solution of doing the inspector only :-)

**Review by RaananW — APPROVED** [review]

**Review by sebavan — APPROVED** [review]

**Pull request merged**